### ISSUE TITLE
MatchboxNet questions

### ISSUE THREAD

**Issue by dmitrytyrin** [NVIDIA/NeMo#2886]
Hello!
Need help with some moments:
1. How to perform GPU batched inference with trained model?
2. Is it possible to make MatchboxNet architecture more heavy? Encountered problem that https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/matchboxnet/matchboxnet_3x1x64_v2.yaml was stucked on ~50% accuracy on clean audio commands (38 labels, trained from scratch, 100 epochs, batch size 512).

**Comment by titu1994** [comment]
An easy way to increase size of matchboxnet is to increase the number of filters in it's intermediate layers (other than it's last layer).

Batches inference can be done with replacing the trainer.train() with trainer.test() and setting up the test_ds in the config instead of train_ds and validation_ds.

**Comment by dmitrytyrin** [comment]
@titu1994, I meant how to perform inference and get predicted labels/logits? trainer.test() returns accuracy and test loss.
In https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Speech_Commands.ipynb I found only CPU inference function.

**Comment by titu1994** [comment]
You can replace the .cpu() with .cuda() and modify the function to push the tensors onto the GPU. It should work exactly the same as CPU mode.
